### ISSUE TITLE
Language Server 1.13.6 crash triggered by `name` field missing from the configuration

### ISSUE THREAD

**Issue by ineffective** [microsoft/vscode-cpptools#10251]
### Environment

- OS and Version: Windows (host) and Linux Debian 9.13
- VS Code Version:
```
Version: 1.74.0 (user setup)
Commit: 5235c6bb189b60b01b1f49062f4ffa42384f8c91
Date: 2022-12-05T16:38:16.075Z (2 days ago)
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19045
Sandboxed: No
```

- C/C++ Extension Version: 1.13.6
- Other extensions you installed (and if the issue persists after disabling them):  ms-vscode.cpptools-extension-pack, twxs.cmake, ms-vscode.cmake-tools, esbenp.prettier-vscode. Disabling seems to do nothing.
- If using SSH remote, specify OS of remote machine:  
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).
Right after staring VSCode and opening remote connection to aforementioned Debian system, information "The language server crashed 5 times in the last 3 minutes. It will not be restarted." is displayed and server isn't restarted.

### Bug Summary and Steps to Reproduce

Bug Summary:

As written in concise bug report. In addition after downgrading to earlier version (previous, in fact: `1.12.4`) everything works again without issues.

### Expected behavior

Language server not crashing 5 times in last 3 minutes :)

### Code sample and Logs

```shell
I can't even get to C/C++ log diagnostics. All I get is:

[Error - 3:20:10 PM] Connection to server got closed. Server will not be restarted.

`c_cpp_properties.json` is as follows (it was reduced thanks to [this issue I had earlier](https://github.com/microsoft/vscode-cpptools/issues/9931)):

{
    "configurations": [
        {
            "intelliSenseMode": "linux-gcc-x64",
            "configurationProvider": "ms-vscode.cmake-tools"
        }
    ],
    "version": 4
}
```

**Comment by ineffective** [comment]
I expected to see something in `~/.vscode-server/data/logs/`, but it seems that there is nothing interesting there - I don't know how anonymous these files are so I won't attach them in bulk, but if anyone is interested in certain files/greps over them I will gladly provide additional info.

**Comment by TessHull** [comment]
We (myself and two colleagues) are seeing the same issue with 1.13.6.
rolling back to 1.12.4 works (except for one of us where their system always updates back to 1.13.6 and won't stay rolled back.)

We are running with version 1.73.1 and the language server is presumably running inside our WSL2 container using the Ubuntu 22.04.1 LTS user-space, which is up-to-date.

**Comment by cdacamar** [comment]
+1 I am also observing the same crashes after updating to 1.13.6.  Since VSCode is my daily, I am forced to revert back to 1.12.4.

**Comment by Colengms** [comment]
Hi @ineffective , @TessHull , @cdacamar .  We'd like to fix this ASAP, but we don't yet have enough information to repro.  There would seem to be something about your environments that we may need to know about in order to investigate further.

Could you enable debug logging using the setting `"C_Cpp.loggingLevel": "Debug"`, and provide everything output to the C/C++ output channel, from launch and up to the crash?

Does the issue occur with a simple example, such as opening a workspace folder with a single empty cpp file?  Might it be isolated to when a cpp file is already open as VS Code launches?

Any additional information you can provide would help.  :)

**Comment by Colengms** [comment]
Hi @ineffective , @TessHull , @cdacamar . Also, could you try the intermediate prerelease versions (1.13.x)?  Isolating which prerelease version introduced the issue could help.

**Comment by ineffective** [comment]
Hi @Colengms, I set output level to `Debug` in all three tabs: `User`, `Workspace` and `Remote [SSH: xc]` (where `xc` is of course my remote machine). Results from various windows are attached. Screenshot shows what as you called them output channels were available. I attached most (all of them) except those that were empty. Regarding entries in `remote-pty-host.txt` - there are tons of errors, but I don't know if this is relevant, as with `1.12.4` entries seem to be nearly identical (of course they would, as up to 2022-12-09 07:41:06 they are same entries) - attached `remote-pty-host-1.12.4.txt`.
Also note, that I used version `1.13.7` to get these logs - I failed to notice that version changed from yesterday.
[remote - ssh.txt](https://github.com/microsoft/vscode-cpptools/files/10192416/remote.-.ssh.txt)
[remote-attach.txt](https://github.com/microsoft/vscode-cpptools/files/10192417/remote-attach.txt)
[remote-extension-host.txt](https://github.com/microsoft/vscode-cpptools/files/10192418/remote-extension-host.txt)
[remote-pty-host.txt](https://github.com/microsoft/vscode-cpptools/files/10192419/remote-pty-host.txt)
[remote-server.txt](https://github.com/microsoft/vscode-cpptools/files/10192420/remote-server.txt)
![sshot-vscode-lang-server-dead-2022-12-09](https://user-images.githubusercontent.com/8870983/206641877-6ea86088-361f-4feb-9068-e30ec1a9c944.png)
[c_cpp window output.txt](https://github.com/microsoft/vscode-cpptools/files/10192422/c_cpp.window.output.txt)
[cpptools.txt](https://github.com/microsoft/vscode-cpptools/files/10192424/cpptools.txt)
[extension-host.txt](https://github.com/microsoft/vscode-cpptools/files/10192425/extension-host.txt)
[json language server.txt](https://github.com/microsoft/vscode-cpptools/files/10192426/json.language.server.txt)
[remote-pty-host-1.12.4.txt](https://github.com/microsoft/vscode-cpptools/files/10192427/remote-pty-host-1.12.4.txt)

**Comment by ineffective** [comment]
Also about installing other versions between `1.12.4` and `1.13.6` - in extension's options I see versions available on the screenshot attached and I have no idea how to install other versions.

![image](https://user-images.githubusercontent.com/8870983/206642345-4e8d13ce-9b0e-4f80-bf41-3dee7e15fe5b.png)

**Comment by ineffective** [comment]
I also tried opening vscode with all source files closed - nothing changes, crash still happens.

After creating directory with only `CMakeFiles.txt`, `src/main.cpp` (empty) and toolchain-file (not empty, but *probably* irrelevant) to mirror basic setup of main project that I have problems with, I used vscode to connect to it and it seems to work. I also added some code (`boost` in use, as it seems to be relatively hard to work with using vscode & extension) and it still works. After restarting vscode it still works, no matter if any source file is opened or closed.

**Comment by ineffective** [comment]
Hi @Colengms, I closed vscode, added to the aforementioned empty project directory .vscode and copied my `c_cpp_properties.json` there, its contents is very simple, it is as follows:

```json
{
    "configurations": [
        {
            "intelliSenseMode": "linux-gcc-x64",
            "configurationProvider": "ms-vscode.cmake-tools"
        }
    ],
    "version": 4
}
```
After opening vscode it crashes nearly immediately. Output streams seem to be basically identical to what I see with original project I had problems with, so I won't attach them - but if you think you need them I will attach them.

**Comment by ineffective** [comment]
I also tried to remove various entries from `c_cpp_properties.json`.
After removing `version` I still got crashes. File was changed by vscode and now contains:
```json
{
    "configurations": [
        {
            "intelliSenseMode": "linux-gcc-x64",
            "configurationProvider": "ms-vscode.cmake-tools",
            "compilerPath": "/usr/bin/gcc",
            "cStandard": "c11",
            "cppStandard": "gnu++14"
        }
    ],
    "version": 4
}
```
Note that in my toolchain file I requested different compiler (`x86_64-linux-gnu-g++-9.2.0`) and c++ standard (`set(CMAKE_CXX_STANDARD 17)`), also I don't think it is right for the vscode to modify this file as it see fit.

I reverted file to what I have shown in previous comment and removed `configurationProvider` (so `version` is back) - it still crashes, but vscode briefly flashed at me `CMake Tools would like to configure IntelliSense for this folder.` I don't think it is relevant though.
I reverted file to what I have shown in previous comment and removed `intelliSenseMode` - still crashes. 
I reverted file once again and left only:
```json
{
    "configurations": [ { } ],
    "version": 4
}
```
Crash still present.
I reverted file once more and left:
```json
{
    "configurations": [  ],
    "version": 4
}
```
(no `{}` in `configurations` array) - I got error `Failed to parse "/home/jd/test/.vscode/c_cpp_properties.json": Invalid configuration file. There must be at least one configuration present in the array.`, probably as expected, but so to say "server failed to crash" - I get only basic code completion (let's say `function based` rather than `context based` - when I write name of an object, dot and try to autocomplete it fails to find anything, but when I write only name of an object and try to autocomplete it shows lambda from same scope that has same prefix) and I can't find include files - it isn't weird as `configurationProvider` is not present and include paths from `CMakeLists.txt` are not picked up (I think). Note that if I don't have `c_cpp_properties.json` at all completion works as expected (so provides list of fields and methods for `<object-name><dot>`).
I reverted config file once again and left only:
```json
{
    "version": 4
}
```
Vscode can't parse configuration file, basically same as above.
File reverted, left only empty object:
```json
{}
```
Again `Failed to parse`.

With empty `c_cpp_properties.json` file it also fails to parse config file.

After completely removing `c_cpp_properties.json` - no crashes.

**Comment by ineffective** [comment]
I can confirm that after adding "name" property everything works as expected.

**Comment by Colengms** [comment]
We'll use this issue to track the repro with missing configuration name.  If anyone is seeing a crash that does not involve a missing configuration name, please open a separate issue.  As mentioned, the work around for this issue is to add a `name` field to the configuration in `c_cpp_properties.json`.

**Comment by zeroxia** [comment]
I do have "name" property in my `c_cpp_properties.json`, and the crash still occurs.
The environment is a docker container running Ubuntu 18.04, and I use vscode to SSH-remotely connect to it.

Ever 10 minutes there is a pop up saying "Connection to server got closed. Server will not be restarted."
And in "Output" pane one "cpptools" window is added, this just keeps adding.

```json
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/ara-api/**",
            ],
            "compilerPath": "/usr/bin/g++",
            "cStandard": "c99",
            "cppStandard": "c++14",
            "intelliSenseMode": "linux-gcc-x64"
        }
    ],
    "version": 4
}
```

I tried to debug `cpptools` with attaching method described here:
https://github.com/microsoft/vscode-cpptools/wiki/Attaching-debugger-to-cpptools-or-cpptools%E2%80%90srv

It is unable to capture the call stack. When crash happens, debugger console window says:
`Program terminated with signal SIGKILL, Killed.`

C/C++ debug log:

[https://pastebin.com/5JGZEq3w](pastebin log)
